src/main/java/com/example/demo/services/ImaggaAPI.java

```diff
@@ -0,0 +1,45 @@
+package com.example.demo.services;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+@Service
+public class ImaggaAPI {
+    private static final String API_KEY = "acc_e978d85bb0300de";
+    private static final String API_SECRET = "cb2a812cb9772d0e275faab050e8f830";
+
+    private final WebClient webClient;
+
+    public ImaggaAPI() {
+        this.webClient = WebClient.create();
+    }
+
+    public String categorizeImage(String imageUrl) {
+        try {
+            String endpointUrl = "https://api.imagga.com/v2/tags";
+            String url = endpointUrl + "?image_url=" + imageUrl;
+
+            return webClient.method(HttpMethod.GET)
+                    .uri(url)
+                    .header(HttpHeaders.AUTHORIZATION, createAuthorizationHeader())
+                    .accept(MediaType.APPLICATION_JSON)
+                    .retrieve()
+                    .bodyToMono(String.class)
```

Comment:
This here can be converted to another class that represents the response body of Imagga. We can use it then to automatically retrieve the labels/